util/pace/util/grid/generation.py

```diff
@@ -409,11 +410,79 @@ def __init__(
             self._calculate_unit_vectors_lonlat = (
                 self._calculate_unit_vectors_lonlat_cube_sphere
             )
-            self._init_dgrid()
-            self._init_agrid()
+            if extdgrid is False:
+                self._init_dgrid()
+                self._init_agrid()
         else:
             raise NotImplementedError(f"Unsupported grid_type = {grid_type}")
 
+    # from_generated class method
+    # Generates a metric terms object, which will perform
+    # the same function as the __init__ method for MetricTerms
+    # initializing the dgrid by input from data contained in an
+    # externally generated tile file
+    @classmethod
+    def from_external(
+        cls,
+        x,
+        y,
+        dx,
+        dy,
+        area,
+        quantity_factory,
+        communicator,
+        grid_type,
+        extdgrid: bool = True,
+    ) -> "MetricTerms":
+        terms = MetricTerms(
```

Comment:
This would be good as a docstring
```suggestion
    @classmethod
    def from_external(
        cls,
        x,
        y,
        dx,
        dy,
        area,
        quantity_factory,
        communicator,
        grid_type,
        extdgrid: bool = True,
    ) -> "MetricTerms":
    """
    Generates a metric terms object, using input from data contained in an
    externally generated tile file
    """
        terms = MetricTerms(
```